server/client.go

```diff
@@ -709,6 +725,7 @@ func (c *client) deliverMsg(sub *subscription, mh, msg []byte) {
 
 	deadlineSet := false
 	if client.bw.Available() < (len(mh) + len(msg) + len(CR_LF)) {
+		atomic.AddInt64(&c.cache.wfc, 1)
```

Comment:
A bit confused here: c.cache belongs to the client 'c' attempting to deliver messages to other clients 'client' subscriptions. The 'client' is the one with a buffer too small, yet we let 'c' know that it should change its buffer size.